akka-cluster/src/main/scala/akka/cluster/Gossip.scala

```diff
@@ -80,6 +80,11 @@ private[cluster] final case class Gossip(
       throw new IllegalArgumentException("Nodes not part of cluster in reachability table, got [%s]"
         format inReachabilityButNotMember.mkString(", "))
 
+    val inReachabilityVersionsButNotMember = overview.reachability.versions.keySet diff members.map(_.uniqueAddress)
+    if (inReachabilityVersionsButNotMember.nonEmpty)
+      throw new IllegalArgumentException("Nodes not part of cluster in reachability versions table, got [%s]"
+        format inReachabilityVersionsButNotMember.mkString(", "))
+
     val seenButNotMember = overview.seen diff members.map(_.uniqueAddress)
     if (seenButNotMember.nonEmpty)
       throw new IllegalArgumentException("Nodes not part of cluster have marked the Gossip as seen, got [%s]"
```

Comment:
Just a thought, with adding a third check, executing similar work with differing inputs, might we want something like
```scala
private def assertInvariants(): Unit = {
  def ifTrueThrow(func: => Boolean, expected: String, actual: String) =
    if (func) throw new IllegalArgumentException(s"$expected, but found [$actual]")

  ifTrueThrow(members.exists(_.status == Removed), s"Live members must not have status [$Removed]", s"${members.filter(_.status == Removed)}")

  val inReachabilityButNotMember = overview.reachability.allObservers diff members.map(_.uniqueAddress)
  ifTrueThrow(inReachabilityButNotMember.nonEmpty, "Nodes not part of cluster in reachability table", inReachabilityButNotMember.mkString(", "))

  val seenButNotMember = overview.seen diff members.map(_.uniqueAddress)
  ifTrueThrow(seenButNotMember.nonEmpty, "Nodes not part of cluster have marked the Gossip as seen", seenButNotMember.mkString(", "))
}
```

Reply:
thanks, incorporated that